src/hotspot/share/gc/shenandoah/shenandoahVerifier.cpp

```diff
@@ -82,7 +94,7 @@ class ShenandoahVerifyOopClosure : public BasicOopIterateClosure {
     T o = RawAccess<>::oop_load(p);
     if (!CompressedOops::is_null(o)) {
       oop obj = CompressedOops::decode_not_null(o);
-
+      obj = ShenandoahForwarding::get_forwardee(obj);
```

Comment:
But... `verify_oop` verifies the consistency of `obj` and its fwdptr. Here, we effectively omit those checks, making verification less effective! Is this for `Reference` classes only?